osf/models/preprint_provider.py

```diff
@@ -79,3 +83,7 @@ def banner_path(self):
             return '/static/img/preprint_providers/{}'.format(self.logo_name)
         else:
             return None
+
+    @property
+    def metaschema(self):
+        return MetaSchema.objects.get(name='Preprint Provider')
```

Comment:
How is this linked to the extra field? Also are there tests (maybe in registrations) that verify that having the metaschema'd field will error appropriately if bad data is submitted? Can you add a test to verify that this field is being appropriately restricted? If there are thorough tests elsewhere, then we just need the one to ensure that the mechanism is active on this field.